fl4health/model_bases/pca.py

```diff
@@ -0,0 +1,166 @@
+from logging import INFO
+from typing import Optional, Tuple
+
+import torch
+import torch.nn as nn
+from flwr.common.logger import log
+from torch import Tensor
+from torch.nn.parameter import Parameter
+
+
+class PCAModule(nn.Module):
+    def __init__(self, low_rank: bool = True, full_svd: bool = False, rank_estimation: int = 6) -> None:
+        """
+        PyTorch module for performing Principal Component Analysis.
+
+        Args:
+            low_rank (bool, optional): Indicates whether the data matrix is low-rank. If the user has
+             prior knowledge that it is, then this parameter can be set to True to allow for more efficient
+              computation of SVD. Defaults to True.
+            full_svd (bool, optional): Indicates whether full SVD or reduced SVD is performed. Defaults to False.
+            rank_estimation (int, optional): An estimation of the rank of the data matrix.
```

Comment:
This will probably be answered below. However, it would be good to have a comment covering it here. Is a rank_estimation value required if `low_rank` is true? I could see a situation where one suspects that the matrix has a low-rank representation but the true rank is unknown. Moreover, what is the "penalty" or "downside" to being off in this estimation?